### PR TITLE
Replace-species

### PR DESCRIPTION
**Pull request by janosh**
`Structure().replace_species` doc string should make it clear that original structure is modified in place.

### CODE REVIEW

File: pymatgen/core/structure.py [hunk 1]
```diff
@@ -1045,28 +1044,30 @@ def matches(self, other, anonymous=False, **kwargs):
...
-            return True
-        if other is None:
-            return False
-        if len(self) != len(other):
-            return False
-        if self.lattice != other.lattice:
-            return False
-        for site in self:
-            if site not in other:
+    def __eq__(self, other: object) -> bool:
```
Comment (shyuep) [inline pymatgen/core/structure.py:1047]: Please do not make this change. I am unclear why your PR is mentioning a replace method but you are modifying the way structure checks for equality.

Comment (janosh) [reply pymatgen/core/structure.py:1047]: Sorry, this wasn't meant to be bunched in with the other small changes in this PR. I'll revert it.

In general, updating all Pymatgen rich comparison functions to send `NotImplemented` follows recommended practice. See https://docs.python.org/3/library/constants.html#NotImplemented. So maybe worth tackling in a separate PR.

Comment (janosh) [reply pymatgen/core/structure.py:1047]: The PR mentioned the `Structure.replace_species()` method cause my original intent was just to update its doc string to make it clearer it makes in-place modifications.

**Comment by mkhorton** [comment]
Thanks @janosh, I see you reverted the other commit so going to merge this

**Pull request merged**